fgpyo/sam/__init__.py

```diff
@@ -843,6 +843,21 @@ def all_recs(self) -> Iterator[AlignedSegment]:
             for rec in recs:
                 yield rec
 
+    def set_tag(
+        self,
+        tag: str,
+        value: Union[str, int, float],
+    ) -> None:
+        """Add a tag to all records associated with the template.
+
+        Args:
+            tag: The name of the tag.
+            value: The value of the tag.
+        """
+
```

Comment:
want to require hat `tag` is two characters?

Reply:
done